app/models/git_repository.rb

```diff
@@ -14,25 +13,31 @@ def initialize(repository_url:, repository_dir:)
 
   def setup!(executor, temp_dir, git_reference)
     executor.output.write("Beginning git repo setup\n")
+    return false unless clone!(executor: executor, from: repository_url, to: repo_cache_dir, mirror: true) unless locally_cached?
     return false unless update!(executor: executor)
     return false unless clone!(executor: executor, from: repo_cache_dir, to: temp_dir)
     return false unless checkout!(executor: executor, pwd: temp_dir, git_reference: git_reference.shellescape)
-    true
+    commit_from_ref(git_reference)
```

Comment:
before `commit_from_ref` needed to be called after setup -> now 1 less thing you have to know, setup just returns the ref or false